onnxruntime/core/providers/qnn/builder/qnn_context_mem_handle_manager.h

```diff
@@ -27,17 +28,18 @@ class QnnContextMemHandleManager {
 
   // Gets an existing QNN mem handle or registers a new one.
   // `qnn_mem_handle` is set to the QNN mem handle and `did_register` is true if `qnn_mem_handle` was newly registered.
-  Status GetOrRegister(void* shared_memory_address, const Qnn_Tensor_t& qnn_tensor,
+  Status GetOrRegister(void* memory_address, bool uses_shared_mem_allocator, const Qnn_Tensor_t& qnn_tensor,
                        Qnn_MemHandle_t& qnn_mem_handle, bool& did_register);
 
-  Status Unregister(void* shared_memory_address);
+  Status Unregister(void* memory_address);
 
   void Clear();
 
  private:
   const QNN_INTERFACE_VER_TYPE& qnn_interface_;
   Qnn_ContextHandle_t context_;
   const logging::Logger& logger_;
+  std::unique_ptr<qnn::RpcMemLibrary> rpc_lib_;
```

Comment:
would prefer to reuse the one that already exists, if possible.

https://github.com/microsoft/onnxruntime/blob/bc7b07dbb41a2f441dbed1a91855563ba0dd8a31/onnxruntime/core/providers/qnn/qnn_execution_provider.h#L105

e.g., a reference to it or shared_ptr copy.

alternatively, maybe we can make this class more generic by updating GetOrRegister() to take the values needed to populate the HTP mem descriptor. then this class doesn't need to be concerned with using HtpSharedMemoryAllocator or RpcMemLibrary to get those values.